tinygrad/tensor.py

```diff
@@ -512,7 +514,7 @@ def batchnorm(self, weight:Optional[Tensor], bias:Optional[Tensor], mean:Tensor,
   def dropout(self, p=0.5) -> Tensor:
     if not Tensor.training: return self
     # TODO: why is this going through numpy?
-    _mask: np.ndarray = np.asarray(Tensor._rng.binomial(1, 1.0-p, size=self.shape), dtype=np.float32)
+    _mask: np.ndarray = np.asarray(Tensor._rng.binomial(1, 1.0-p, size=self.shape), dtype=Tensor.default_type.np)
```

Comment:
This should be refactored to not use numpy directly.

Reply:
@geohot done

Reply:
@geohot though now it looks like this could have been a separate PR